src/Kestrel.Transport.Sockets/SocketConnection.cs

```diff
@@ -223,7 +244,13 @@ private void SetupSendBuffers(ReadableBuffer buffer)
             }
             finally
             {
+                // Now, complete the input so that no more reads can happen
+                Input.Complete(error ?? new ConnectionAbortedException());
```

Comment:
Do we need to do this here? It happens on the input side. It's a bit cleaner than the libuv transport.

Reply:
https://github.com/aspnet/KestrelHttpServer/blob/dev/test/Kestrel.FunctionalTests/RequestTests.cs#L1081-L1091

If we don't complete the input with a ConnectionAbortedException here prior to initiating a connection shutdown, the test will close the client half of the connection which will complete the input gracefully.

If something is still trying to read after the output is closed, that's a problem. The connection is aborted at this point. The exact same code is there for libuv: https://github.com/aspnet/KestrelHttpServer/blob/dev/src/Kestrel.Transport.Libuv/Internal/LibuvConnection.cs#L72-L82

Reply:
The feature is fine I just want it to happen at the "right place" which IMO is the end of the input loop. 

https://github.com/aspnet/KestrelHttpServer/blob/dev/src/Kestrel.Transport.Sockets/SocketConnection.cs#L208-L215

Reply:
I think you linked to the end of the output loop. Just like the LibuvConnection, Sockets completes the input both when the output is closed and the input is closed. This means input is always completed twice in both transports.

Reply:
I meant this:

https://github.com/aspnet/KestrelHttpServer/blob/dev/src/Kestrel.Transport.Sockets/SocketConnection.cs#L111-L122

Reply:
I see. Well, it's removed now and we're back to one call to IPipeWriter.Complete(). I'm wondering if we should change the libuv transport as well.

Reply:
It's easier here since there's 2 async loops

Reply:
True. If the libuv output loop exits first, it will call uv_close which prevents the read callback from receiving notification of the disconnect.